### ISSUE TITLE
Mitmweb: accessibility: Screenreaders have a tough time with the MITMweb UI

### ISSUE THREAD

**Issue by zersiax** [mitmproxy/mitmproxy#4463]
#### Problem Description
Screenreaders depend on semantic HTML to do their best work. The way mitmweb works is divergent from the norm in various ways, which means WAI-ARIA can and should be used in order to make the UI usable for screenreader users. The current issues:

- It is currently impossible to know what flow has focus, this is seemingly only indicated visually.
- The shortcut key list (triggered by ? ) seems to be using an inaccessible editor component and therefore cannot be used. Likely codeMirror or a CM derivative of some kind. That is, if I'm seeing the right thing, it took me several tries to get it to come up.
- The keyboard shortcut help (invoked with ?) I haven't been able to get to work at all.
- Obviously there might be more, but these are currently the biggest showstoppers.
#### Steps to reproduce the behavior:
1. Enable any kind of screen reader (Narrator or NVDA on Windows, VoiceOver on Mac, Orca on Linux).
2. Arrow through the flows in an MTIMmweb session.
3. Observe that nothing happens whatsoever.

#### Suggested Remediation Steps:

- Add a live region ( aria-live="polite") to the interface.
- On  changing flow, set the inner HTML of the live region to some distinguishing flow identifier, think ID, host name etc. This will make the screenreader echo the new content and should make arrowing through flows work at least.
- nice-to-have: echo brief status messages to that same region to indicate other operations have happened, e.g. flow info having been opened and other such things.
- Semantically mark up the various regions of the UI. Right now it's an absolute pain to get from the list of flows to ...anywhere, really. Even adding invisible headings would be better than what it's doing now.

#### System Information
Paste the output of "mitmproxy --version" here.
Mitmproxy: 5.3.0
Python:    3.6.9
OpenSSL:   OpenSSL 1.1.1h  22 Sep 2020
Platform:  Linux-4.15.0-117-generic-x86_64-with-Ubuntu-18.04-bionic

**Comment by mhils** [comment]
Hi @zersiax,

Sorry for letting you down here and thanks for speaking up. I think I wrote most of mitmweb at a point where I didn't have much experience with web accessibility. We definitely should do better here.

FWIW I don't think we have a keyboard shortcut help, but we do use CodeMirror, so that might be what's causing it. Let's check if we can upgrade to CodeMirror 6 on that front, which claims to be much more accessible.

**Comment by zersiax** [comment]
oh wow :)
I seem to have made a bit of a mistake here :) I was under the impression the "user interface" area of the docs were about mitmweb but they are not, which ...explains a thing or two about why things weren't working as expected :)
A quick update:

- CodeMirror is indeed accessible in the sense that the contents of the editor cannot be read. This means you can't check input, can't see what's already in the editor without crl+a ctrl+c-ing it into another editor. There are more accessible alternatives, chiefly the Monaco editor. I don't know about the latest version of codeMirror in that regard, but can certainly find out.
- ignore the bit about arrowing through flows, that doesn't apply in the web version. However, adding a bit of ARIA to indicate stuff has happened would be good. Essentially, the table rows work when you hit enter on them, but this isn't indicated to a screenreader very clearly which is really easy to fix.

I'll report back into this issue when I find more, and see if I can't fix some of these myself in a PR later :)

**Comment by mhils** [comment]
Fantastic, thanks! 🍰 If you have any questions please don't hesistate dropping by in our developer Slack. I unfortunately won't be able to spend any significant time on mitmweb in the next few weeks, but this here is definitely something I'm going to look at once I get there.

[CodeMirror 6](https://codemirror.net/6/) claims to be "designed from the start with accessibility in mind, avoiding a number of pitfalls that the current crop of editors suffer from" as their very top feature, so I'd be cautiously optimistic here. Then again I'd also be happy to switch to Monaco, we really don't do that much with the editor at the moment. :)

**Comment by prabhatjoshi321** [comment]
May I take on this Issue
